tests/ui/explicit-tail-calls/two-phase.rs

```diff
@@ -0,0 +1,11 @@
+// regression test for <https://github.com/rust-lang/rust/issues/112788>
```

Comment:
```suggestion
// regression test for <https://github.com/rust-lang/rust/issues/112788>
// This test used to ICE because we tried to run drop glue of `x`
// if dropping `_y` (happening at the `become` site) panicked and caused an unwind.
```